lib/wireprotocol/command.js

```diff
@@ -24,7 +24,12 @@ function command(server, ns, cmd, options, callback) {
   const readPreference = getReadPreference(cmd, options);
 
   let finalCmd = Object.assign({}, cmd);
-  if (isSharded(server) && readPreference && readPreference.preference !== 'primary') {
+  if (
+    isSharded(server) &&
+    !supportsOpMsg(server) &&
```

Comment:
nit: pull the `supportsOpMsg` call into a const higher up so it could be reused at line 58